### PR TITLE
docs(lambda): document adding execution permissions to provided IAM roles

### PR DESCRIPTION
**Pull request by flemjame-at-amazon**
### Commit Message
docs(lambda): document adding execution permissions to provided IAM roles

If I am providing a Role for a Lambda function, it currently isn't given the basic execution permissions, so the function cannot log anything or, in the case of a VPC Lambda, it cannot create the network interfaces. The user has to add those permissions themselves, but it isn't clear from the documentation that that needs to happen.

This commit adds documentation showing CDK users how to add the required permissions for execution.
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by flemjame-at-amazon** [comment]
> What is the exact use case you are trying to achieve here?

This doesn't unlock any new use cases -- this was intended to make Lambda creation foolproof.

> There are users of this library who want to have full control over the permissions of their role. The property `role` is provided to satisfy this.

I get the full control portion. However, without the basic execution role, or equivalent permissions,  Lambda does not work correctly (cannot log output). In a VPC, the Lambda does not even initialize without the EC2 permissions. Is this the desired outcome for users who want full control?

At the very least, it should be clearly documented that those permissions must be added yourself when providing your own role. I've hit that issue, as have co-workers, which was the impetus for this PR.

**Comment by nija-at** [comment]
> I get the full control portion. However, without the basic execution role, or equivalent permissions, Lambda does not work correctly (cannot log output). In a VPC, the Lambda does not even initialize without the EC2 permissions. Is this the desired outcome for users who want full control?

How about a `LambdaRole` class that extends `iam.Role` which comes with the basic permissions? Does that provide a good starting point for customers with this use case?

File: packages/@aws-cdk/aws-lambda/README.md [hunk 1]
```diff
@@ -49,6 +49,37 @@ to our CDK project directory. This is especially important when we want to share
...
 different techniques for bundling resources into libraries.
 
+### Execution Role
+
+Lambda functions assume an IAM role during execution. In CDK by default, Lambda
+functions will use an autogenerated Role if one is not provided.
+
+The autogenerated Role is automatically given permissions to execute the Lambda
+function. To reference the autogenerated Role:
+```
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/README.md:59]: New line and hint for syntax highlighting

````suggestion

```ts
````

File: packages/@aws-cdk/aws-lambda/README.md [hunk 2]
```diff
@@ -49,6 +49,37 @@ to our CDK project directory. This is especially important when we want to share
...
+  handler: 'index.handler',
+  code: lambda.Code.fromAsset(path.join(__dirname, 'lambda-handler')),
+  
+fn.role // the Role
+```
+
+You can also provide your own IAM role. Provided IAM roles will not automatically
+be given permissions to execute the Lambda function. To provide a role and grant
+it appropriate permissions:
+```
```
Comment (nija-at) [inline packages/@aws-cdk/aws-lambda/README.md:71]:
````suggestion

```ts
````

**Comment by flemjame-at-amazon** [comment]
> Nice! Me love some good clean documentation and writing. Some minor adjustments called out below.
> 
> Let me know if you think it's a good idea and would like to pursue the `LambdaRole` option in this PR.

I like the idea -- it's just a Role with the "assumedBy" always being Lambda. 

Once concern I have is I'd prefer not to create a class like "LambdaRoleProps" because I worry about keeping it's features in sync with regular RoleProps. Do you have any ideas as to how to achieve this?

**Comment by nija-at** [comment]
> because I worry about keeping it's features in sync with regular RoleProps

What if `LambdaRoleProps` extended `RoleProps`? Would that work?

**Pull request merged**